File: frontend/quiz-plateform/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';
import { Observable } from 'rxjs';
import { Quiz } from '../models/quiz';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private http: HttpClient) { }

  getAllQuizzes(): Observable<Quiz[]> {
    return this.http.get<Quiz[]>(`${baseUrl}/quizzes`);
  }

  getQuizById(id: number): Observable<Quiz> {
    return this.http.get<Quiz>(`${baseUrl}/quizzes/${id}`);
  }

  getQuizzesByTest(testId: number): Observable<Quiz[]> {
    return this.http.get<Quiz[]>(`${baseUrl}/quizzes/test/${testId}`);
  }

  createQuiz(quiz: Quiz): Observable<Quiz> {
    return this.http.post<Quiz>(`${baseUrl}/quizzes`, quiz);
  }

  updateQuiz(id: number, quiz: Quiz): Observable<Quiz> {
    return this.http.put<Quiz>(`${baseUrl}/quizzes/${id}`, quiz);
  }

  deleteQuiz(id: number): Observable<void> {
    return this.http.delete<void>(`${baseUrl}/quizzes/${id}`);
  }

  getQuizByTitre(titre: string): Observable<Quiz> {
    return this.http.get<Quiz>(`${baseUrl}/quizzes/titre/${titre}`);
  }
}

File: backend/quizhire/quizhire/src/main/java/com/quizhire/quizhire/services/QuizService.java
package com.quizhire.quizhire.services;

import java.util.List;

import com.quizhire.quizhire.models.test.Quiz;
import com.quizhire.quizhire.models.test.Test;

public interface QuizService {
    List<Quiz> getQuizzes();
    Quiz getQuizById(Integer id) throws Exception;
    List<Quiz> getQuizzesByTest(Test test);
    Quiz addQuiz(Quiz quiz);
    Quiz updateQuiz(Integer id, Quiz quiz) throws Exception;
    void deleteQuiz(Integer id) throws Exception;
    Quiz findbyTitre(String titre);
}

